cmd/main.go

```diff
@@ -194,7 +194,8 @@ func runBuild(ctx context.Context, args []string) error {
 
 func getCaddyOutputFile() string {
 	f := "." + string(filepath.Separator) + "caddy"
-	if runtime.GOOS == "windows" {
+	// compiling on or for Windows, use .exe extension
+	if runtime.GOOS == "windows" || os.Getenv("GOOS") == "windows" {
```

Comment:
Does this make sense?

```suggestion
	if os.Getenv("GOOS") == "windows" || (os.Getenv("GOOS") == "" && runtime.GOOS == "windows") {
```

Reply:
Yeah, but `cmp.Or` is more straightforward. But it's not available in 1.21 😅